src/object.d

```diff
@@ -3241,6 +3241,23 @@ template RTInfo(T)
     enum RTInfo = null;
 }
 
+private int __cmp(F)(const F f1, const F f2)
+if (__traits(isFloating, F))
+{
+    static if (is(F == cfloat) || is(F == cdouble) || is(F == creal))
+    {
+        // Use rt.cmath2._Ccmp instead ?
+        auto r = __cmp(f1.re, f2.re);
+        if (!r) r = __cmp(f1.im, f2.im);
+        return r;
+    }
+    else static if (is(F == float) || is(F == double) || is(F == real))
+    {
+        return (f1 > f2) - (f2 > f1);
```

Comment:
This generates nice code with dmd and gdc, and also much faster than the previous version (and also a bit faster than the obvious alternatives).